kafka-ui-react-app/src/components/Brokers/Broker/Configs/Configs.tsx

```diff
@@ -34,14 +34,13 @@ const Configs: React.FC = () => {
 
   const getData = () => {
     return data
-      .filter(
-        (item) =>
-          item.name.toLocaleLowerCase().indexOf(keyword.toLocaleLowerCase()) >
-          -1
-      )
+      .filter((item) => {
+        const nameMatch = item.name.toLocaleLowerCase().includes(keyword);
+        const valueMatch = item.value && item.value.includes(keyword);
```

Comment:
you can actually check the `nameMatch` before doing value check , so it wouldn't do more work if it does not have to.

Reply:
Actually i don't get it what you are saying. Do you mean that if nameMatch found then i don't need to check for valueMatch ?

Reply:
> Actually i don't get it what you are saying. Do you mean that if nameMatch found then i don't need to check for valueMatch ?

Yep, I think this is correct, and I think the suggestion is to do something like
```
      .filter((item) => {
        const nameMatch = item.name.toLocaleLowerCase().includes(keyword);
        return nameMatch ? nameMatch : item.value && item.value.includes(keyword); // try to match the keyword on any of the item.value elements when nameMatch fails but item.value exists
      })
```

or more explicitly:
```
      .filter((item) => {
        const nameMatch = item.name.toLocaleLowerCase().includes(keyword);
        if (nameMatch === true) return nameMatch;
        const valueMatch = item.value && item.value.includes(keyword);
        return valueMatch;
      })
```
instead of (the current implementation):
```
      .filter((item) => {
        const nameMatch = item.name.toLocaleLowerCase().includes(keyword);
        const valueMatch = item.value && item.value.includes(keyword);
        return nameMatch || valueMatch;
      })
```

so we don't need to go through the entire `item.value` array in case we find a `nameMatch` (and `item.value` exists).